src/overlap_check/ProgressBar.cpp

```diff
@@ -0,0 +1,68 @@
+
+
+#include "ProgressBar.hpp"
+
+#include <sstream>
+#include <iostream>
+#include <iomanip>
+
+#if defined(__unix__) || defined(__unix) || (defined(__APPLE__) && defined(__MACH__))
+#include <unistd.h>
+#endif
+
+#define BAR_WIDTH 72
+
+ProgressBar::ProgressBar() {
+  // initialize bar
+  set_value(0.0);
+}
+
+ProgressBar::~ProgressBar() {
+  std::cout << "\n";
```

Comment:
Suggest replacing single line in destructor with the following:
```c++
  if (need_final_newline) {
    std::cout << "\n";
  }
```
This is to remove superfluous newline on successful completion [part 2 of 4].